src/Orleans.CodeGenerator/FieldIdAssignmentHelper.cs

```diff
@@ -0,0 +1,207 @@
+                var isConstructorParameter = false;
+
+                if (!id.HasValue)
+                {
+                    var property = GetMatchingProperty(field);
+                    if (property is null)
+                    {
+                        continue;
+                    }
+
+                    id = CodeGenerator.GetId(_libraryTypes, property);
+                    if (!id.HasValue)
+                    {
+                        var constructorParameter = _constructorParameters.FirstOrDefault(x => x.Name.Equals(property.Name, StringComparison.OrdinalIgnoreCase));
+                        if (constructorParameter is not null)
+                        {
+                            id = (ushort)_constructorParameters.IndexOf(constructorParameter);
+                            isConstructorParameter = true;
+                        }
+                    }
+                }
+
+                if (id.HasValue)
+                {
+                    _symbols[member] = (id.Value, isConstructorParameter);
+                }
+            }
+        }
+        return true;
+    }
+
+    private IPropertySymbol? GetMatchingProperty(IFieldSymbol field)
+    {
+        var propertyName = Regex.Match(field.Name, "^<([^>]+)>.*$");
+        if (!propertyName.Success)
+        {
+            return null;
+        }
+
+        var name = propertyName.Groups[1].Value;
+        var candidates = _memberSymbols.OfType<IPropertySymbol>()
+            .Where(property => string.Equals(name, property.Name, StringComparison.Ordinal)
+                               && SymbolEqualityComparer.Default.Equals(field.Type, property.Type)).ToArray();
+        return candidates.Length == 1 ? candidates[0] : null;
+    }
+
+    private IFieldSymbol? GetMatchingField(IPropertySymbol property)
+    {
+        var backingFieldName = $"<{property.Name}>k__BackingField";
+        var candidates = (from field in _memberSymbols.OfType<IFieldSymbol>()
+                          where SymbolEqualityComparer.Default.Equals(field.Type, property.Type)
+                          where field.Name == backingFieldName || GetCanonicalName(field.Name) == GetCanonicalName(property.Name)
+                          select field).ToArray();
+        return candidates.Length == 1 ? candidates[0] : null;
+    }
+
+    private string GetCanonicalName(string name)
+    {
+        name = name.TrimStart('_');
+        if (name.Length > 0 && char.IsUpper(name[0]))
+            name = $"{char.ToLowerInvariant(name[0])}{name.Substring(1)}";
+        return name;
+    }
+
+    private (string, ushort) GetCanonicalNameAndFieldId(ITypeSymbol typeSymbol, string name)
+    {
+        name = GetCanonicalName(name);
+
+        // compute the hash from the type name (without namespace, to allow it to move around) and name
+        var hashCode = (2617 * typeSymbol.ToDisplayString(SymbolDisplayFormat.MinimallyQualifiedFormat).GetHashCode())
```

Comment:
This will not produce a stable hash code, since string has a randomized (per process) hash code algorithm

Reply:
I force-pushed a commit to use xxhash instead, PTAL

Reply:
My commit broke things with a FileNotFoundException for the System.IO.Hashing dll

Reply:
This is a common problem people have with source generators, due to the mixing of target frameworks. I caught up on some of the issue discussions on GitHub, but was not able to solve it using any of the suggested workarounds (lots of project file fiddling to get the unit test project to load the netstandard2.0 assembly).

Instead, I grabbed and embedded the relevant source files for the Xx32 hash implementation. I changed the namespace and made the classes `internal`. The source is MIT licensed so I think it's okay to copy like this (at worst, it'll be Microsoft stealing from itself, so probably still ok). We can switch to `System.IO.Hashing` when the toolchain is fixed (i.e. unified to run on netcore6).